content/docs/support/troubleshooting.md

```diff
@@ -51,6 +54,52 @@ Diagnostic logging can also be controlled with flags and environment variables o
 $ TF_LOG=TRACE pulumi up --logtostderr --logflow -v=10 2> out.txt
 ```
 
+## Attaching a debugger to a Pulumi program
+
+When developing or troubleshooting Pulumi programs, you may need to debug your code. This example walks you through starting a Node.js debugging session in VS Code and setting breakpoints.
+
+### Create VS Code launch configuration
+
+In your project directory, create or update `.vscode/launch.json` with the following configuration to enable debugging:
+
+```json
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Launch Pulumi Program (debug)",
+            "type": "node",
+            "request": "attach",
+            "continueOnAttach": true,
+            "skipFiles": [
+                "${workspaceFolder}/node_modules/**/*.js",
+                "${workspaceFolder}/lib/**/*.js",
+                "<node_internals>/**/*.js"
+            ]
+        }
+    ]
+}
+```
+
+### Set breakpoints and start Pulumi in debug mode
+
+Open your program and set breakpoints by clicking in the gutter next to the line numbers.
+
+In a terminal run `NODE_OPTIONS=”--inspect-brk” pulumi up` to start the deployment process in debug mode. This will cause Pulumi to pause execution and wait for a debugger to attach.
+
+Press `F5` in VS Code to start debugging. VS Code will attach to the waiting Node.js process, and execution will continue until reaching the first breakpoint you have set.
+
+For a step-by-step guide for attaching a debugger to a Pulumi program, check out this [blog on breakpoint debugging.](/blog/next-level-iac-breakpoint-debugging/)
+
+### Debugging by each SDK
+
+To attach a debugger across Pulumi's supported SDKs, you can configure the program to wait for the debugger to attach before continuing execution.
+
+- Golang: Use `dlv exec` command to run the program, attach by port. Build using debug flags.
+- Python: Use `debugpy` in entrypoint script to listen and wait, attach by port. Must install debugpy into venv.
+- Csharp: Poll System.Diagnostics.Debugger.IsAttached` in entrypoint script to wait, attach by pid.
+- Java: Use `jdwp` JVM args, attach by port.
+
```

Comment:
For this section, have we tested these instructions?

Reply:
removed for now